core/src/main/scala/cats/data/Prod.scala

```diff
@@ -75,30 +75,27 @@ private[data] sealed abstract class ProdInstances4 extends ProdInstances5 {
 }
 
 private[data] sealed abstract class ProdInstances5 extends ProdInstances6 {
-  implicit def catsDataMonadForProd[F[_], G[_]](implicit FM: Monad[F], GM: Monad[G]): Monad[λ[α => Prod[F, G, α]]] = new ProdMonad[F, G] {
-    def F: Monad[F] = FM
-    def G: Monad[G] = GM
+  implicit def catsDataApplyForProd[F[_], G[_]](implicit FF: Apply[F], GG: Apply[G]): Apply[λ[α => Prod[F, G, α]]] = new ProdApply[F, G] {
+    def F: Apply[F] = FF
+    def G: Apply[G] = GG
   }
 }
 
 private[data] sealed abstract class ProdInstances6 extends ProdInstances7 {
+  implicit def catsDataFunctorForProd[F[_], G[_]](implicit FF: Functor[F], GG: Functor[G]): Functor[λ[α => Prod[F, G, α]]] = new ProdFunctor[F, G] {
+    def F: Functor[F] = FF
+    def G: Functor[G] = GG
+  }
   implicit def catsDataFoldableForProd[F[_], G[_]](implicit FF: Foldable[F], GF: Foldable[G]): Foldable[λ[α => Prod[F, G, α]]] = new ProdFoldable[F, G] {
     def F: Foldable[F] = FF
     def G: Foldable[G] = GF
   }
 }
 
-private[data] sealed abstract class ProdInstances7 extends ProdInstances8 {
-  implicit def catsDataTraverseForProd[F[_], G[_]](implicit FF: Traverse[F], GF: Traverse[G]): Traverse[λ[α => Prod[F, G, α]]] = new ProdTraverse[F, G] {
-    def F: Traverse[F] = FF
-    def G: Traverse[G] = GF
-  }
-}
-
-private[data] sealed abstract class ProdInstances8 {
-  implicit def catsDataMonadCombineForProd[F[_], G[_]](implicit FF: MonadCombine[F], GF: MonadCombine[G]): MonadCombine[λ[α => Prod[F, G, α]]] = new ProdMonadCombine[F, G] {
-    def F: MonadCombine[F] = FF
-    def G: MonadCombine[G] = GF
+private[data] sealed abstract class ProdInstances7 {
+  implicit def catsDataAlternativeForProd[F[_], G[_]](implicit FF: Alternative[F], GG: Alternative[G]): Alternative[λ[α => Prod[F, G, α]]] = new ProdAlternative[F, G] {
```

Comment:
Thanks! We are getting close. 
`Alternative` should be between `MonadCombine` and  `MonoidK`, 
`Monad` should be between `MonadCombine` and `Applicative` 
how about we do
`ProdInstances`: `MonadCombine`, `Order`, `Show` ,`Traverse`, `Contravariant`
`ProdInstances0`: `Alternative`, `Monad`, `Eq`,  `Foldable`
`ProdInstances1`: `MonoidK`, `Applicative`
`ProdInstances2`: `SemigroupK`, `Apply`
`ProdInstances3`: `Functor`

I might make some mistakes here too. 
You may find this diagram helpful 
https://camo.githubusercontent.com/28086d65ea5688b722ef4c19ee70e1974c5f4cb0/687474703a2f2f766972656f2e6f72672f746d702f74797065636c61737365732e737667